src/sharing/ShareToPm.js

```diff
@@ -54,7 +54,13 @@ const styles = createStyleSheet({
 });
 
 type Props = $ReadOnly<{|
-  ...$Exact<NavigationNavigatorProps<{||}, {| params: {| sharedData: SharedData |} |}>>,
+  // Since we've put this screen in a tab-nav route config, it gets the
+  // `navigation` prop for free.
+  navigation: NavigationTabProp<{|
+    ...NavigationStateRoute,
+    params: {| sharedData: SharedData |},
+  |}>,
```

Comment:
Hmm -- for some reason it seems like this type annotation is not working (it's resulting in `any` types), whereas in at least some spot-checking the ones from the previous commit with `NavigationStackProp` are working.

Specifically, if I go and do something silly like
```diff
   state = (() => {
-    const { sharedData } = this.props.navigation.state.params;
+    const { sharedData, asdfasdf } = this.props.navigation.state.params;
```

then I don't get any type error at the attempt to pull an `asdfasdf` property from `params`.

Also, hovering on the identifier `navigation` in that line I just quoted shows "any(implicit)" as the type, and ditto for the subsequent `state` or `params`. (That's the symptom I first noticed; I tried the `asdfasdf` thing just to confirm that it wasn't some IDE glitch.)

Perhaps relatedly, if in VS Code I hit F12 on `NavigationTabProp` to try to go to its definition, it doesn't find it -- it can only take me to the `import type` line at the top of this file. Whereas it does work fine for `NavigationStackProp` in a file like `AccountDetailsScreen.js`. There *does* seem to be a libdef in `flow-typed/`, so not sure what the issue is.

Reply:
I have a hunch that I'll look into tomorrow—we've got two libdefs for `react-navigation-tabs` in `flow-typed/npm`, and one of them is a stub where everything is `any`.

Reply:
Aha...that's it. Yeah, getting rid of that stub libdef (which we don't want anyway) reveals lots of Flow errors that seem productive to address.

Reply:
Aha! Good catch.